### PR TITLE
[WIP/RFC] Extract generic methods from `WorkerNode` and `Cluster` classes.

### PR DESCRIPTION
**Pull request by fnordahl**
Note that this PR depends on and includes commits from #173

### CODE REVIEW

File: ovn-tester/ovn_workload.py [hunk 1]
```diff
@@ -216,6 +196,123 @@ def wait(self, sbctl, timeout_s):
...
+    @ovn_stats.timeit
+    def provision_port(self, cluster, passive=False):
+        raise NotImplemented
+
+    @ovn_stats.timeit
+    def ping_external(self, cluster, port):
+        raise NotImplemented
+
+
+class WorkerNode(ChassisNode):
```
Comment (fnordahl) [inline ovn-tester/ovn_workload.py:284]: @dceara it is this inheritance that made me think of `ChassisNode` as `Compute` is not really fit with Kubernetes. Or what do you think?

Comment (fnordahl) [reply ovn-tester/ovn_workload.py]: It could also serve as a base for a `NetworkNode`

**Comment by fnordahl** [comment]
To better understand how this could be used I've also created a OpenStack CMS plugin and test skeleton here: https://github.com/fnordahl/ovn-heater/commit/9e254e9471d52bb05d1cdf5706272305fbc514b0

File: ovn-tester/ovn_workload.py (hunk 1 above)
Comment (dceara) [reply ovn-tester/ovn_workload.py]: I'm ok with both; if `ChassisNode` makes more sense to you let's go with that. :)

**Comment by fnordahl** [comment]
The patches from this PR has been carried over to #179, let's review them there.

**Pull request closed without merging**